src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -98,239 +102,151 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes)
+
         /// <summary>
-        /// Sonoma API only provides Tensor() constructors with overloaded
-        /// versions based on data type.
+        /// Creates a NamedOnnxValue from a scalar value.
         /// </summary>
-
-        private static Dictionary<System.Type, DataType> _typeMap;
-
-        public static Tensor CreateScalarTensor<T>(T data)
+        /// <typeparam name="T">The type of the Tensor contained in the NamedOnnxValue</typeparam>
+        /// <param name="name">The name of the NamedOnnxValue</param>
+        /// <param name="data">The data values of the Tensor</param>
+        /// <returns>NamedOnnxValue</returns>
+        public static NamedOnnxValue CreateScalarNamedOnnxValue<T>(string name, T data)
         {
-            if (typeof(T) == typeof(System.Boolean))
-            {
-                return new Tensor((System.Boolean)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Byte))
-            {
-                return new Tensor((System.Byte)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Char))
-            {
-                return new Tensor((System.Char)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Double))
-            {
-                return new Tensor((System.Double)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Single))
-            {
-                return new Tensor((System.Single)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Int32))
-            {
-                return new Tensor((System.Int32)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Int64))
-            {
-                return new Tensor((System.Int64)(object)data);
-            }
-            else if (typeof(T) == typeof(System.SByte))
-            {
-                return new Tensor((System.SByte)(object)data);
-            }
-            else if (typeof(T) == typeof(System.Int16))
-            {
-                return new Tensor((System.Int16)(object)data);
-            }
-            else if (typeof(T) == typeof(System.UInt32))
-            {
-                return new Tensor((System.UInt32)(object)data);
-            }
-            else if (typeof(T) == typeof(System.UInt64))
-            {
-                return new Tensor((System.UInt64)(object)data);
-            }
-            else if (typeof(T) == typeof(System.UInt16))
-            {
-                return new Tensor((System.UInt16)(object)data);
-            }
-            throw new NotSupportedException($"Unsupported type {typeof(T)}");
+            var typeMap = SystemTypeToOnnxType();
+            if (!typeMap.ContainsKey(typeof(T)))
+                throw new NotImplementedException($"Not implemented type {typeof(T)}");
+            return NamedOnnxValue.CreateFromTensor<T>(name, new DenseTensor<T>(new T[] { data }, new int[] { }));
```

Comment:
The empty array for the `dimensions` - `new int[] { }` - I don't think is correct. That is going to throw an exception.  See [https://github.com/dotnet/corefx/blob/d76c2039f22a2607967f894a83f1041fad77d8e9/src/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs#L284-L287](https://github.com/dotnet/corefx/blob/d76c2039f22a2607967f894a83f1041fad77d8e9/src/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs#L284-L287)

Reply:
Good catch. Replaced with 0 instead of empty.

---
In reply to: [236343398](https://github.com/dotnet/machinelearning/pull/1717#discussion_r236343398) [](ancestors = 236343398)

Reply:
Shouldn't it be `1`?

---
In reply to: [236476609](https://github.com/dotnet/machinelearning/pull/1717#discussion_r236476609) [](ancestors = 236476609,236343398)

Reply:
Yes -- sorry, it should be 1! -- not sure why it was empty to begin with (hence replaced with zero). If it contains value, it needs to have a positive dimension.